src/ng/directive/input.js

```diff
@@ -1593,19 +1589,16 @@ function rangeInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     setInitialValueAndObserver(maxAttrType, 'max', maxChange);
   }
 
-  function setInitialValueAndObserver(attrType, attrName, changeFn) {
-    var initialAttrValue;
-    if (attrType === attrName) {
-      initialAttrValue = attr[attrName];
-      // Set the actual element attribute so that the browser can adjust the value based on
-      // the max value, which might be interpolated
-      element.attr(attrName, initialAttrValue);
+  function setInitialValueAndObserver(actualAttrName, htmlAttrName, changeFn) {
+    // e.g. max === max
+    if (actualAttrName === htmlAttrName) {
+      // interpolated attributes set the attribute value only after a digest, but we need the
+      // attribute value when the input is first rendered, so that the browser can adjust the
+      // input value based on the min/max value
+      element.attr(htmlAttrName, attr[htmlAttrName]);
```

Comment:
Iirc, `ngMin/Max` will set `attrs.min/max`, but not the element's attribute or property. The reason is that we want to be able to have in-Angular validation, but not validation from the browser. This makes sense for other inputs, but not for `input[range]`, because it cannot have invalid values (in supporting browsers at least).

So, in a sense, the browser forces its restrictions on the user (they can't enter a value that is outside of the specified range). Because of that, I agree that `ngMin/Max` don't have anything to offer for `input[range]`.

I would add a little warning in the docs, though, saying that `ngMin/Max` will hav no effect on `input[range]`, because I expect many people will try to use them :smile: